ir/src/integrity_stmts/graph.rs

```diff
@@ -467,9 +490,10 @@ pub enum Operation {
     /// value is the length of the column's periodic cycle. The periodic value made available from
     /// the specified column is based on the current row of the trace.
     PeriodicColumn(usize, usize),
-    /// A random value provided by the verifier. The inner value is the index of this random value
-    /// in the array of all random values.
-    RandomValue(usize),
+    /// A random value provided by the verifier. The first inner value is the index of this random
+    /// value in the array of all random values and the second is value index of random value in
+    /// case it represented by array, e.g. `rand: [a, b[4]]`.
+    RandomValue(usize, usize),
```

Comment:
Let's keep this the way it was. The new `RandomValue` syntax should just be semantic sugar. We should handle most of the changes during parsing, and the final part (identifying `Expression::Elem` and `Expression::VectorAccess` as random values) can be done here in the IR before getting the index in the full flattened array and saving it as an `Operation::RandomValue(usize)` type